File: app/src/main/java/com/example/dropex/fragments/OrdersFragment.java
package com.example.dropex.fragments;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.GridView;

import com.example.dropex.R;
import com.example.dropex.adpaters.CartGVAdapter;
import com.example.dropex.adpaters.OrderGVAdapter;
import com.example.dropex.enums.Size;
import com.example.dropex.models.CartModel;
import com.example.dropex.models.OrdersModel;

import java.util.ArrayList;

public class OrdersFragment extends Fragment {

    GridView orderGVAdapter;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_orders, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        orderGVAdapter = view.findViewById(R.id.ordersGV);
        ArrayList<OrdersModel> ordersModelArrayList = new ArrayList<>();

        ordersModelArrayList.add(new OrdersModel(1, 2, Size.l, 1, 2));
        ordersModelArrayList.add(new OrdersModel(1, 1, Size.m, 1, 3));
        ordersModelArrayList.add(new OrdersModel(1, 1, Size.xxl, 1, 4));

        OrderGVAdapter adapter = new OrderGVAdapter(view.getContext(), ordersModelArrayList);
        orderGVAdapter.setAdapter(adapter);
    }
}

File: app/src/main/java/com/example/dropex/fragments/OrdersFragment.kt
package com.example.dropex.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.GridView
import android.widget.ProgressBar
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.lifecycle.lifecycleScope
import com.example.dropex.R
import com.example.dropex.adpaters.OrderGVAdapter
import com.example.dropex.database.OrdersRepository
import com.example.dropex.database.ProductsRepository
import com.example.dropex.models.ProductModel
import kotlinx.coroutines.launch

class OrdersFragment : Fragment() {
    private lateinit var orderGV: GridView
    private var ordersRepository: OrdersRepository = OrdersRepository()
    private var productsRepository: ProductsRepository = ProductsRepository()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_orders, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        orderGV = view.findViewById(R.id.ordersGV)
        val loadingBar = view.findViewById<ProgressBar>(R.id.progressBar)
        val noItemsFoundTV = view.findViewById<TextView>(R.id.noItemsFoundTV)

        orderGV.visibility = View.GONE
        loadingBar.visibility = View.VISIBLE

        viewLifecycleOwner.lifecycleScope.launch {
            val ordersModelArrayList = ordersRepository.getOrders()
            if(ordersModelArrayList.size == 0) {
                loadingBar.visibility = View.GONE
                noItemsFoundTV.visibility = View.VISIBLE
                return@launch
            }
            val productModelArrayList = ArrayList<ProductModel?>()
            for (cartModel in ordersModelArrayList) {
                productModelArrayList.add(productsRepository.getProductById(cartModel!!.productId))
            }
            val adapter = OrderGVAdapter(view.context, ordersModelArrayList, productModelArrayList)
            orderGV.adapter = adapter

            orderGV.visibility = View.VISIBLE
            loadingBar.visibility = View.GONE
            noItemsFoundTV.visibility = View.GONE
        }
    }
}